go/api/response_handlers.go

```diff
@@ -31,6 +31,15 @@ func handleStringOrNullResponse(response *C.struct_CommandResponse) string {
 	return handleStringResponse(response)
 }
 
+func handleStringOrNullResponseWithFree(response *C.struct_CommandResponse) string {
```

Comment:
There is already a handleStringOrNullResponse which is handling the same functionality. Please let me know in case I am missing anything.

Reply:
@janhavigupta007 since it's already been merge, can you take care of it in a separate PR?

Ill let @MikeMwita and @eifrah-aws to answer the specific point.

Reply:
@janhavigupta007 
I could refactor the code by combining the two to a single  handles both the memory management and the string conversion in one place like this :
```
func handleStringOrNullResponse(response *C.struct_CommandResponse) string {
	if response == nil {
		return ""
	}
	// need to free the memory 
	defer C.free_command_response(response)

	return convertCharArrayToString(response.string_value, response.string_value_len)
}

```
The major difference lies in memory management . I needed to free the memory after processing the response by calling C.free_command_response(response)  . So I was avoiding memory leaks .

Reply:
`handleStringOrNullResponse` is invoking `handleStringResponse` if the response is not null which invokes the `defer C.free_command_response(response)` which should free up the command_response based on my understanding.

Reply:
Yeah . The handleStringOrNullResponse existing implementation doesnt do this .

Reply:
Sorry my understanding is that `handleStringOrNullResponse` is handling the memory management as well. Can you help me understand in which scenario it would be unable to clean up the memory?